git/init.go

```diff
@@ -111,5 +112,38 @@ func Init(c *Client, opts InitOptions, dir string) (*Client, error) {
 	}
 	c.WorkDir = WorkDir(wd)
 	c.GitDir = GitDir(wd + "/.git")
+
+	if opts.Template.String() != "" {
+		err = filepath.Walk(opts.Template.String(), func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+			path, err = filepath.Rel(opts.Template.String(), path)
+			if err != nil {
+				return err
+			}
+			if info.IsDir() {
+				os.MkdirAll(filepath.Join(c.GitDir.String(), path), 0777)
+			} else {
+				newFile, err := os.Create(filepath.Join(c.GitDir.String(), path))
+				if err != nil {
+					return err
+				}
+				defer newFile.Close()
+				f, err := os.Open(filepath.Join(opts.Template.String(), path))
+				if err != nil {
+					return err
+				}
+				defer f.Close()
+				_, err = io.Copy(newFile, f)
+				if err != nil {
+					return err
+				}
+			}
+
+			return nil
+		})
+	}
```

Comment:
still needs a `return c, err` after the `})` and before the `}`, otherwise the err is being discarded.

(This is the only blocker of my comments, the rest are just to make it a little more idiomatic.)